danielMunteanu/src/main/java/danielCollectionTask/AbstractUser.java

```diff
@@ -0,0 +1,36 @@
+package danielCollectionTask;
+
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class AbstractUser {
+    final String username;
+    final String id;
+    private Map<String, Address> addresses;
+
+    public AbstractUser(String id, String username, Map<String, Address> addresses) {
+        this.id = id;
```

Comment:
the id should be random generated

Reply:
Get rid of 'id' as a parameter in constructor (it refers, as well, to the classes that extend AbstractUser)

Reply:
fixed